diff --git a/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h b/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h
index ffe89bc..40e0c25 100644
--- a/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h
+++ b/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h
@@ -38,6 +38,7 @@
 #ifndef FCL_SHAPE_CONVEX_INL_H
 #define FCL_SHAPE_CONVEX_INL_H
 
+#include <cassert>
 #include <map>
 #include <set>
 #include <utility>
