diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(SQLite3 REQUIRED)
-find_package(Boost REQUIRED COMPONENTS filesystem system thread)
+find_package(Boost REQUIRED COMPONENTS filesystem thread)
 find_package(class_loader REQUIRED)
 find_package(OpenSSL REQUIRED)
 find_package(rclcpp REQUIRED)
@@ -35,6 +35,7 @@ set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 14)
 
 target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME} PUBLIC
+  Boost::headers
   class_loader::class_loader
   rclcpp::rclcpp
   warehouse_ros::warehouse_ros
@@ -65,11 +66,11 @@ if(BUILD_TESTING)
 
   ament_add_gtest(TestLoader test/TestLoader.cpp)
   target_link_libraries(TestLoader ${PROJECT_NAME})
-  target_link_libraries(TestLoader ${Boost_LIBRARIES})
+  target_link_libraries(TestLoader Boost::filesystem Boost::thread)
 
   ament_add_gtest(BusyHandler test/BusyHandler.cpp)
   target_link_libraries(BusyHandler ${PROJECT_NAME})
-  target_link_libraries(BusyHandler ${Boost_LIBRARIES})
+  target_link_libraries(BusyHandler Boost::filesystem Boost::thread)
 
   find_package(ament_lint_auto REQUIRED)
   list(APPEND AMENT_LINT_AUTO_EXCLUDE
