diff --git a/CMakeLists.txt b/CMakeLists.txt
index dfed661..4f7473e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -259,21 +259,33 @@ target_link_libraries(decentralized_multirobot_slam_toolbox_node decentralized_m
 add_library(async_slam_toolbox src/slam_toolbox_async.cpp)
 target_link_libraries(async_slam_toolbox toolbox_common kartoSlamToolbox ${Boost_LIBRARIES})
 add_executable(async_slam_toolbox_node src/slam_toolbox_async_node.cpp)
+if(MSVC)
+  target_compile_options(async_slam_toolbox_node PRIVATE "/F 40000000")
+endif()
 target_link_libraries(async_slam_toolbox_node async_slam_toolbox)
 
 add_library(sync_slam_toolbox src/slam_toolbox_sync.cpp)
 target_link_libraries(sync_slam_toolbox toolbox_common kartoSlamToolbox ${Boost_LIBRARIES})
 add_executable(sync_slam_toolbox_node src/slam_toolbox_sync_node.cpp)
+if(MSVC)
+  target_compile_options(sync_slam_toolbox_node PRIVATE "/F 40000000")
+endif()
 target_link_libraries(sync_slam_toolbox_node sync_slam_toolbox)
 
 add_library(localization_slam_toolbox src/slam_toolbox_localization.cpp)
 target_link_libraries(localization_slam_toolbox toolbox_common kartoSlamToolbox ${Boost_LIBRARIES})
 add_executable(localization_slam_toolbox_node src/slam_toolbox_localization_node.cpp)
+if(MSVC)
+  target_compile_options(localization_slam_toolbox_node PRIVATE "/F 40000000")
+endif()
 target_link_libraries(localization_slam_toolbox_node localization_slam_toolbox)
 
 add_library(lifelong_slam_toolbox src/experimental/slam_toolbox_lifelong.cpp)
 target_link_libraries(lifelong_slam_toolbox toolbox_common kartoSlamToolbox ${Boost_LIBRARIES})
 add_executable(lifelong_slam_toolbox_node src/experimental/slam_toolbox_lifelong_node.cpp)
+if(MSVC)
+  target_compile_options(lifelong_slam_toolbox_node PRIVATE "/F 40000000")
+endif()
 target_link_libraries(lifelong_slam_toolbox_node lifelong_slam_toolbox)
 
 add_library(map_and_localization_slam_toolbox src/experimental/slam_toolbox_map_and_localization.cpp)
